### PR TITLE
Url read and write between Explore and Route

### PR DESCRIPTION
**Pull request by RCAVelez**
Able to send and read trip object between explore and route views in the url
Image selection is persistent between views
Markers are persistent and editable between views

### CODE REVIEW

File: frontend/src/ExploreView.js [hunk 1]
```diff
@@ -44,29 +63,18 @@ function Explore() {
...
-  );
-
   return (
     <div className={styles.exploreContainer}>
       <div className={styles.attractionsSection}>
         <div className={styles.attractionImages}>
-          {allAttractions.map((attraction) => (
-            <div className={styles.attractionContainer} key={attraction.id}>
+          {allAttractions.map((attraction, index) => (
+            <div className={styles.attractionContainer} key={index}>
```
Comment (zoebarth) [inline frontend/src/ExploreView.js:77]: Why did you switch from id to index?

File: frontend/src/ExploreView.js [hunk 2]
```diff
@@ -185,19 +179,18 @@ function Explore() {
    * @param {string} photoUrl photo url
```
Comment (zoebarth) [inline frontend/src/ExploreView.js:179]: Make sure to update @param tags

File: frontend/src/ExploreView.js (hunk 1 above)
Comment (Cindyzhang977) [reply frontend/src/ExploreView.js:77]: we got rid of `id` since it wasn't used for anything

**Comment by zoebarth** [comment]
Make sure you resolve all merge conflicts, I know it's your favorite :)

File: frontend/src/RouteView.js [hunk 3]
```diff
@@ -10,28 +12,42 @@ import OptimizeButton from './route/OptimizeButton.js';
...
+      setAttractions(optimizedOrder);
     }
   }, [isOptimized, optimizedOrder]);
 
   async function optimize() {
     if (!optimizedOrder) {
       const response = await fetch('/api/v1/optimize', {
         method: 'POST',
-        body: JSON.stringify({ attractions: places }),
+        body: JSON.stringify({ attractions: attractions }),
```
Comment (mecolmg) [inline frontend/src/RouteView.js:50]: nit: just `{ attractions }`

File: frontend/src/ExploreView.js (hunk 1 above)
Comment (RCAVelez) [reply frontend/src/ExploreView.js:77]: ^

File: frontend/src/ExploreView.js (hunk 2 above)
Comment (RCAVelez) [reply frontend/src/ExploreView.js:179]: done

**Comment by RCAVelez** [comment]
> Make sure you resolve all merge conflicts, I know it's your favorite :)

will do

File: frontend/src/ExploreView.js [hunk 4]
```diff
@@ -8,21 +8,40 @@ import { useLocation, useHistory } from 'react-router-dom';
...
-  const [centerLocation, setCenterLocation] = useState({});
-  const [allAttractions, setAllAttractions] = useState([]);
-  const [tripId, setTripId] = useState('');
-  const [tripName, setTripName] = useState('');
   const urlParameters = useLocation();
   const query = getQueryParameters(urlParameters.search);
-  const [searchText, setSearchText] = useState(query.search || '');
+  const searchText = query.search || '';
+  const [tripObject, setTripObject] = useState(
+    Object.prototype.hasOwnProperty.call(query, 'trip')
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/ExploreView.js:15]: yup, changed to trip in query

File: frontend/src/ExploreView.js [hunk 5]
```diff
@@ -163,18 +150,25 @@ function Explore() {
    * @return {object[]} array of all attractions
    */
   function getAllAttractions(attractions) {
-    const allAttractions = [];
+    const newAllAttractions = [];
     for (const attraction of attractions) {
       if (Object.prototype.hasOwnProperty.call(attraction.photos[0], 'getUrl')) {
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/ExploreView.js:155]: done

File: frontend/src/ExploreView.js [hunk 6]
```diff
@@ -163,18 +150,25 @@ function Explore() {
...
         const attractionName = attraction.name;
         const photoUrl = attraction.photos[0].getUrl();
         const latLng = attraction.geometry.location;
-        const id = attraction.id;
-        const newAttraction = createAttraction(attractionName, id, latLng, photoUrl);
-        allAttractions.push(newAttraction);
+        const isSelected = selectedAttractions.some(
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/ExploreView.js:159]: Credit to Cindy!

File: frontend/src/RouteView.js [hunk 7]
```diff
@@ -10,28 +12,42 @@ import OptimizeButton from './route/OptimizeButton.js';
...
-import { MOCK_DATA } from './route/mockData.js';
-
 /**
  * Render the route page with list of locations in order and directions on a map between the locations.
  */
 function RouteView() {
   const [isOptimized, setIsOptimized] = useState(false);
   const [isSaved, setIsSaved] = useState(false);
-  const [places, setPlaces] = useState(MOCK_DATA);
+  // const [places, setPlaces] = useState(MOCK_DATA);
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/RouteView.js:21]: removed, done

File: frontend/src/RouteView.js (hunk 3 above)
Comment (RCAVelez) [reply frontend/src/RouteView.js:50]: done

File: frontend/src/RouteView.js [hunk 8]
```diff
@@ -10,28 +12,42 @@ import OptimizeButton from './route/OptimizeButton.js';
...
+
+  /**
+   * Extract the url parameters and convert to dictionary
+   * @param {string} query url string
+   * @return {object} key value pair of url parameters
+   */
+  function getQueryParameters(query) {
+    const params = query.split('?')[1];
+    return Object.fromEntries(new URLSearchParams(params));
+  }
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/RouteView.js:38]: made util file and imported to both Explore and Route

File: frontend/src/ExploreView.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/ExploreView.js:155]: done

File: frontend/src/ExploreView.js [hunk 9]
```diff
@@ -8,21 +8,40 @@ import { useLocation, useHistory } from 'react-router-dom';
...
       if (status === google.maps.places.PlacesServiceStatus.OK) {
         const latLng = results[0].geometry.location;
         const coordinates = new google.maps.LatLng(latLng.lat(), latLng.lng());
-        setCenterLocation({ lat: coordinates.lat(), lng: coordinates.lng() });
+        const newTripObject = JSON.parse(JSON.stringify(tripObject));
+        newTripObject.centerLocation = {
+          lat: coordinates.lat(),
+          lng: coordinates.lng(),
+        };
+        setTripObject(newTripObject);
```
[the remark this answers is not in the document]
Comment (RCAVelez) [reply frontend/src/ExploreView.js:43]: Ah so thats how to do it. done

**Pull request merged**